File: src/stage_game.c
#include "stage_game.h"

void stage_game_init(struct stage_game *this)
{
    /* Initialize virtual member functions */
    this->stage.update = stage_game_update;
    this->stage.render = stage_game_render;

    /* Initialize member variables */
    this->test = 0;
}

void stage_game_fini(struct stage_game *this)
{
    UNUSED(this);
}

void stage_game_update(struct stage *this)
{
    /* TEMPORARY */
    UNUSED(this);
}

void stage_game_render(struct stage *this)
{
    /* TEMPORARY */
    UNUSED(this);
}

File: stage_game.go
package main

import "fmt"

type StageGame struct {
	test int
}

func (stage StageGame) Update() {
	fmt.Println("StageGame Update()")
}

func (stage StageGame) Render() {
	fmt.Println("StageGame Render()")
}

func (stage StageGame) HandleEvents(e *Engine) {
}
